daemon/cmd/kube_proxy_replacement.go

```diff
@@ -334,6 +334,16 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		}
 	}
 
+	if option.Config.InstallNoConntrackIptRules {
+		// InstallNoConntrackIptRules can only be enabled when Cilium is
+		// running in full KPR mode as otherwise conntrack would be
+		// required for NAT operations
+		if !option.Config.KubeProxyReplacementFullyEnabled() {
+			option.Config.InstallNoConntrackIptRules = false
+			log.Infof("%s requires the agent to run in full KPR mode. Disabling it", option.InstallNoConntrackIptRules)
```

Comment:
```suggestion
			log.Infof("%s requires the agent to run with %s=%s. Disabling it.", option.InstallNoConntrackIptRules, option.KubeProxyReplacement, option.KubeProxyReplacementStrict)
```

Reply:
I don't think this is strictly correct as even with `--kube-proxy-replacement probe` it's possible to enable `--install-no-conntrack-iptables-rules` if all the KPR features get enabled

Reply:
Ah, right. I'm a bit worried the current version of the error message isn't actionable by users though. Maybe we should advice to run with `kube-proxy-replacement=strict` without stating the feature *requires* that? It's probably best if users move from `probe` to `strict` anyway if they can.

Reply:
Makes sense :+1: Let's stick with your original suggestion